include/zephyr/drivers/i3c.h

```diff
@@ -0,0 +1,1705 @@
+	 * - Set All Addresses to Static Addresses (SETAASA)
+	 * - Set New Dynamic Address (SETNEWDA)
+	 * - Set Dynamic Address from Static Address (SETDASA)
+	 *
+	 * 0 if address has not been assigned.
+	 */
+	uint8_t dynamic_addr;
+
+#ifdef CONFIG_I3C_USE_GROUP_ADDR
+	/**
+	 * Group address for this target device. Set during:
+	 * - Reset Group Address(es) (RSTGRPA)
+	 * - Set Group Address (SETGRPA)
+	 *
+	 * 0 if group address has not been assigned.
+	 */
+	uint8_t group_addr;
+#endif /* CONFIG_I3C_USE_GROUP_ADDR */
+
+	/**
+	 * Bus Characteristic Register (BCR)
+	 * - BCR[7:6]: Device Role
+	 *   - 0: I3C Target
+	 *   - 1: I3C Controller capable
+	 *   - 2: Reserved
+	 *   - 3: Reserved
+	 * - BCR[5]: Advanced Capabilities
+	 *   - 0: Does not support optional advanced capabilities.
+	 *   - 1: Supports optional advanced capabilities which
+	 *        can be viewed via GETCAPS CCC.
+	 * - BCR[4}: Virtual Target Support
+	 *   - 0: Is not a virtual target.
+	 *   - 1: Is a virtual target.
+	 * - BCR[3]: Offline Capable
+	 *   - 0: Will always response to I3C commands.
+	 *   - 1: Will not always response to I3C commands.
+	 * - BCR[2]: IBI Payload
+	 *   - 0: No data bytes following the accepted IBI.
+	 *   - 1: One data byte (MDB, Mandatory Data Byte) follows
+	 *        the accepted IBI. Additional data bytes may also
+	 *        follows.
+	 * - BCR[1]: IBI Request Capable
+	 *   - 0: Not capable
+	 *   - 1: Capable
+	 * - BCR[0]: Max Data Speed Limitation
+	 *   - 0: No Limitation
+	 *   - 1: Limitation obtained via GETMXDS CCC.
+	 */
+	uint8_t bcr;
+
+	/**
+	 * Device Characteristic Register (DCR)
+	 *
+	 * Describes the type of device. Refer to official documentation
+	 * on what this number means.
+	 */
+	uint8_t dcr;
+
+	struct {
+		/** Maximum Read Speed */
+		uint8_t maxrd;
+
+		/** Maximum Write Speed */
+		uint8_t maxwr;
+
+		/** Maximum Read turnaround time in microseconds. */
+		uint32_t max_read_turnaround;
+	} data_speed;
+
+	struct {
+		/** Maximum Read Length */
+		uint16_t mrl;
+
+		/** Maximum Write Length */
+		uint16_t mwl;
+
+		/** Maximum IBI Payload Size. Valid only if BCR[2] is 1. */
+		uint8_t max_ibi;
+	} data_length;
+
+	/** Private data by the controller to aid in transactions. Do not modify. */
+	void *controller_priv;
+
+#ifdef CONFIG_I3C_USE_IBI
+	/**
+	 * In-Band Interrupt (IBI) callback.
+	 */
+	i3c_target_ibi_cb_t ibi_cb;
+#endif /* CONFIG_I3C_USE_IBI */
+};
+
+/**
+ * @brief Structure initializer for i3c_device_desc from devicetree
+ *
+ * This helper macro expands to a static initializer for a <tt>struct
+ * i3c_device_desc</tt> by reading the relevant bus and device data
+ * from the devicetree, and an associated IBI callback.
+ *
+ * @param node_id Devicetree node identifier for the I3C device whose
+ *                struct i3c_device_desc to create an initializer for.
+ * @param ibi_cb Pointer to IBI callback.
+ */
+#define I3C_DEVICE_DESC_DT_WITH_IBI_CB(node_id, ibi_cb)			\
```

Comment:
we've been using the dt-spec pattern, so would be nice to use the same terminology here for consistency, comment applies to other initializers in this file, would probably be nice to have `_dt` APIs as well (if they apply)

Reply:
The dt-spec pattern in I2C kind of implies that you can simply create a struct at runtime and pass it to the API. However, it cannot be done the same way here. The device descriptor struct is mutable at runtime, that it must be created at build time and its pointer must be passed to APIs. Therefore I am trying not to use the same terminology to avoid confusion. Though `_DT_` is still there to indicate it has some properties are coming from devicetree. I am fine either way, so please let me know your thought on this.